cpu/stm32/periph/timer.c

```diff
@@ -149,28 +148,29 @@ int timer_set_absolute(tim_t tim, int channel, unsigned int value)
 
 int timer_set(tim_t tim, int channel, unsigned int timeout)
 {
+    unsigned value = (dev(tim)->CNT + timeout) & timer_config[tim].max;
```

Comment:
You want to move that below the `irq_disable()`

Reply:
I debated doing that before submitting the PR. I didn't feel strongly either way, but my reasoning for making it as it is follows. Please check my thinking for flaws.

I reasoned that from a caller's stand point, the value of `timeout` is an offset from "now". "Now" is a moving target until `value` is computed, therefore it should be computed as soon as possible, but it doesn't have to be in the critical section. There is no harm in servicing IRQs or higher priority threads between calculating `value` and scheduling the timer to fire once it reaches `value`, so long as the timer doesn't go past `value` + one rollover period of the timer. If the caller needs this guarantee, I think the IRQs need disabled at a higher level then any part of the body of this function can provide.

If this line were moved to the critical section, this function could sill be blocked before entering the critical section. Once unblocked and inside the critical section `value` could be computed, but it would be not what the caller expected. It would be "now" + however long the blocked time was before the critical section. Note that my current implementation also has this flaw too since an it can be blocked between being called and the first line of c code of the body of the function.

In both cases, if the caller is worried about interrupts and other threads messing with setting a timer, the caller better handle it, because the driver will do something bad either way. So best to just keep the critical section as small as possible.

Reply:
As I understand it. The critical sections in this driver exists so that the read-modify-writes on the timers' registers and `_oneshot` are safely synchronized between the timer's ISR and any concurrent threads accessing the driver. The critical sections aren't meant to ensure any kind of timing/blocking guarantees.

Reply:
That makes sense - and we do have a check for `value > timeout` when in the meantime the timeout already passed, so this is indeed a valid solution.